src/Template/TemplateSection.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace League\Plates\Template;
+
+/**
+ * Defines the object format of the data target 
+ * to be rendered in a Template's Section.
+ */
+class TemplateSection
+{
+    /**
+     * Section mode.
+     *
+     * @var int
+     */
+    protected $mode;
+
+    /**
+     * Section content.
+     *
+     * @var string
+     */
+    protected $content;
+
+    /**
+     * Section name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @param string  $name
+     * @param ?string $content
+     * @param ?int    $mode
+     */
+    public function __construct(string $name, $content = '', $mode = Template::SECTION_MODE_REWRITE)
+    {
+        $this->name = $name;
+        $this->content = $content;
+        $this->mode = $mode;
+    }
+
+    /**
+     * @param  string $content
+     * @param  int    $mode
+     * @return void
+     */
+    public function add(string $content, int $mode)
+    {
+        $this->mode = $mode;
+
+        // if this template doesn't have that section, so we just add it.
+        if (empty($this->content)) {
```

Comment:
I like !isset instead of empty. Some time ago I read that this would be faster.

Reply:
I think it's because empty also checks if the value represents an empty value like `""`.